Fix build with exiv2 0.28.
https://github.com/LuminanceHDR/LuminanceHDR/commit/e5ebd48fac2de33f4990f530c68dd4c9d8fc3829
(CMakeLists.txt minimum-version hunk omitted)
--- src/Exif/ExifOperations.cpp
+++ src/Exif/ExifOperations.cpp
@@ -42,6 +42,14 @@
 #include "ExifOperations.h"
 #include "arch/math.h"
 
+#if EXIV2_TEST_VERSION(0,28,0)
+typedef Exiv2::Error Exiv2Error;
+typedef Exiv2::Image::UniquePtr ImagePtr;
+#else
+typedef Exiv2::AnyError Exiv2Error;
+typedef Exiv2::Image::AutoPtr ImagePtr;
+#endif
+
 using namespace boost;
 using namespace boost::assign;
 
@@ -108,7 +116,7 @@ void copyExifData(const std::string &from, const std::string &to,
 #endif
 
     try {
-        Exiv2::Image::AutoPtr sourceImage;
+        ImagePtr sourceImage;
         Exiv2::ExifData srcExifData;
 
         if (!from.empty()) {
@@ -128,7 +136,7 @@ void copyExifData(const std::string &from, const std::string &to,
         }
 
         // get destination exif data
-        Exiv2::Image::AutoPtr destinationImage = Exiv2::ImageFactory::open(to);
+        ImagePtr destinationImage = Exiv2::ImageFactory::open(to);
 
         if (dontOverwrite) {
             // doesn't throw anything if it is empty
@@ -212,7 +220,7 @@ void copyExifData(const std::string &from, const std::string &to,
             destinationImage->setExifData(srcExifData);
         }
         destinationImage->writeMetadata();
-    } catch (Exiv2::AnyError &e) {
+    } catch (Exiv2Error &e) {
 #ifndef NDEBUG
         qDebug() << e.what();
 #endif
@@ -250,7 +258,7 @@ float obtain_avg_lum(const std::string& filename)
 {
     try
     {
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open(filename);
+        ImagePtr image = Exiv2::ImageFactory::open(filename);
         image->readMetadata();
         Exiv2::ExifData &exifData = image->exifData();
         if (exifData.empty())
@@ -329,7 +337,7 @@ allowed for ev computation purposes.
             return -1;
         }
     }
-    catch (Exiv2::AnyError& e)
+    catch (Exiv2Error& e)
     {
         return -1;
     }
@@ -338,7 +346,7 @@ allowed for ev computation purposes.
 
 float getExposureTime(const std::string &filename) {
     try {
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open(filename);
+        ImagePtr image = Exiv2::ImageFactory::open(filename);
         image->readMetadata();
         Exiv2::ExifData &exifData = image->exifData();
         if (exifData.empty()) return -1;
@@ -374,14 +382,14 @@ float getExposureTime(const std::string &filename) {
         } else {
             return -1;
         }
-    } catch (Exiv2::AnyError &e) {
+    } catch (Exiv2Error &e) {
         return -1;
     }
 }
 
 float getAverageLuminance(const std::string &filename) {
     try {
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open(filename);
+        ImagePtr image = Exiv2::ImageFactory::open(filename);
         image->readMetadata();
         Exiv2::ExifData &exifData = image->exifData();
 
@@ -403,7 +411,7 @@ float getAverageLuminance(const std::string &filename) {
                   << std::endl;
 
         return -1.0;
-    } catch (Exiv2::AnyError &e) {
+    } catch (Exiv2Error &e) {
         return -1.0;
     }
 }
--- src/Libpfs/exif/exifdata.cpp
+++ src/Libpfs/exif/exifdata.cpp
@@ -25,6 +25,16 @@
 #include <exiv2/exiv2.hpp>
 #include <iostream>
 
+#if EXIV2_TEST_VERSION(0,28,0)
+typedef Exiv2::Error Exiv2Error;
+typedef Exiv2::Image::UniquePtr ImagePtr;
+#define EXIV2_TO_INT toInt64
+#else
+typedef Exiv2::AnyError Exiv2Error;
+typedef Exiv2::Image::AutoPtr ImagePtr;
+#define EXIV2_TO_INT toLong
+#endif
+
 namespace pfs {
 namespace exif {
 
@@ -52,7 +62,7 @@ ExifData::ExifData(const std::string &filename) { fromFile(filename); }
 void ExifData::fromFile(const std::string &filename) {
     reset();
     try {
-        ::Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open(filename);
+        ::ImagePtr image = Exiv2::ImageFactory::open(filename);
         image->readMetadata();
         ::Exiv2::ExifData &exifData = image->exifData();
 
@@ -121,7 +131,7 @@ void ExifData::fromFile(const std::string &filename) {
          */
         if ((it = exifData.findKey(Exiv2::ExifKey("Exif.Image.Orientation"))) !=
             exifData.end()) {
-            long rotation = it->toLong();
+            long rotation = it->EXIV2_TO_INT();
             switch (rotation) {
                 case 3:
                     m_orientation = 180;
@@ -134,7 +144,7 @@ void ExifData::fromFile(const std::string &filename) {
                     break;
             }
         }
-    } catch (Exiv2::AnyError &e) {
+    } catch (Exiv2Error &e) {
         return;
     }
 }
--- src/TransplantExif/TransplantExifDialog.cpp
+++ src/TransplantExif/TransplantExifDialog.cpp
@@ -27,14 +27,11 @@
 #include <exif.hpp>
 #include <image.hpp>
 #include <exiv2/exiv2.hpp>
-// Make sure an EXIV2_TEST_VERSION macro exists:
-#ifdef EXIV2_VERSION
-# ifndef EXIV2_TEST_VERSION
-# define EXIV2_TEST_VERSION(major,minor,patch) \
-    ( EXIV2_VERSION >= EXIV2_MAKE_VERSION(major,minor,patch) )
-# endif
+
+#if EXIV2_TEST_VERSION(0,28,0)
+typedef Exiv2::Error Exiv2Error;
 #else
-# define EXIV2_TEST_VERSION(major,minor,patch) (false)
+typedef Exiv2::AnyError Exiv2Error;
 #endif
 
 #include "Common/config.h"
@@ -347,7 +344,7 @@ void TransplantExifDialog::transplant_requested() {
                 QFile::encodeName((*i_dest)).constData(),
                 m_Ui->checkBox_dont_overwrite->isChecked());
             m_Ui->rightlist->item(index)->setBackground(QBrush("#a0ff87"));
-        } catch (Exiv2::AnyError &e) {
+        } catch (Exiv2Error &e) {
             add_log_message("ERROR:" + QString::fromStdString(e.what()));
             m_Ui->rightlist->item(index)->setBackground(QBrush("#ff743d"));
         }
